src/rtc.rs

```diff
@@ -12,25 +36,73 @@ use crate::{
     rcc::Rcc,
 };
 
-/// Entry point to the RTC API
-pub struct RTC {
+#[doc(no_inline)]
+pub use rtcc::{Datelike, NaiveDate, NaiveDateTime, NaiveTime, Timelike};
+
+/// Errors that can occur when dealing with the RTC.
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    /// Invalid input data was used (e.g. a year outside the 2000-2099 range).
+    InvalidInputData,
+}
+
+/// Binary coded decimal with 2 bytes.
+struct Bcd2 {
+    pub tens: u8,
+    pub units: u8,
+}
+
+/// Two 32-bit registers (RTC_TR and RTC_DR) contain the seconds, minutes, hours
+/// (12- or 24-hour format), day (day of week), date (day of month), month, and
+/// year, expressed in binary coded decimal format (BCD). The sub-seconds value
+/// is also available in binary format.
+///
+/// The following helper functions encode into BCD format from integer and
+/// decode to an integer from a BCD value respectively.
+fn bcd2_encode(word: u32) -> Result<Bcd2, Error> {
+    let tens = (word / 10)
+        .try_into()
+        .map_err(|_| Error::InvalidInputData)?;
+    let units = (word % 10)
+        .try_into()
+        .map_err(|_| Error::InvalidInputData)?;
+    Ok(Bcd2 { tens, units })
+}
+
+fn bcd2_decode(first: u8, second: u8) -> u32 {
+    (first * 10 + second).into()
+}
+
+/// Entry point to the RTC API.
+pub struct Rtc {
     rtc: pac::RTC,
     read_twice: bool,
 }
 
-impl RTC {
-    /// Initializes the RTC API
+impl Rtc {
+    /// Initializes the RTC API.
+    ///
+    /// The `init` argument will only be used, if the real-time clock is not
+    /// already configured. If the clock is not yet configured, and init is set
+    /// to `None`, then the datetime corresponding to `2000-01-01 00:00:00`
+    /// will be used for initialization.
     ///
-    /// The `initial_instant` argument will only be used, if the real-time clock
-    /// is not already configured. Its `subsecond` field will be ignore in any
-    /// case.
+    /// # Errors
+    ///
+    /// Returns [`Error::InvalidInputData`](Error::InvalidInputData) if the
```

Comment:
```suggestion
    /// Returns [`Error::InvalidInputData`] if the
```

That should also work, and is a bit simpler.